### DISCUSSION TITLE
Sendind file with json-rpc (multipart form-data)

### DISCUSSION THREAD

**Discussion by kvelaro** [yoanm/symfony-jsonrpc-http-server#83] [Q&A]
Sending file is not working in /json-rpc (cause multipart form data). Is there any workaround?

**Comment by yoanm** [comment]
Hello 👋 

Interesting question indeed.
By keeping in mind the fact that JSON-RPC is not tied at all to HTTP (specifications are defined regardless of the transport protocol, which may be just TCP, so without any kind of headers like with HTTP), I would say that using multipart form data is more likely not the right way in that case.
By looking on web, a simple workaround might be to encode the file content (see [there](https://stackoverflow.com/questions/5835500/could-i-use-json-rpc-to-post-a-file-to-a-webservice) for instance) and send it as part of the JSON-RPC request.

However, you may reach a point where the file to send is so large that it doesn't work due to memory or CPU limitation on server side (as the request payload is entirely loaded in memory when it is decoded from JSON).
In that case, without having thoroughly checked on web, I see few ways to explore: 
 - the simplest, just don't use JSON-RPC at all, add an HTTP endpoint instead (like you would do with any symfony app)
 - Or implement a mechanism allowing to send chunks to the JSON-RPC method, so whatever the file size, it won't be an issue on server side.

Feel free to let me know which way you picked (or if you found another 😉)
